emcc.py

```diff
@@ -1206,7 +1206,8 @@ def phase_setup(state):
   else:
     target = 'a.out.js'
 
-  settings.TARGET_BASENAME = unsuffixed_basename(target)
+  settings.TARGET_BASENAME_WITH_EXT = os.path.basename(target)
```

Comment:
Does your code assume this file is main the javascript file?  Which case I would call this something more like `TARGET_JS_FILE` and delay assigning it until later down when we actual calculate the JS filename (see `js_target = `..  in `phase_final_emitting`..  I would just set it then to the same value is `js_target`).

Reply:
> Does your code assume this file is main the javascript file?

Yes, but it needs to be relative so that it's usable in `import` expression, not the full path. If anything, I'd rename `TARGET_BASENAME` instead as it seems misleading - `basename` normally means "just the filename" (with extension included), whereas in Emscripten `TARGET_BASENAME` has extension already stripped.

> and delay assigning it until later down when we actual calculate the JS filename

Hm, is it ever going to be different than just basename?

Reply:
> so assuming that TARGET is JavaScript at this point is not correct.

Ohh true, I thought this was JS already. Good point, I'll fix that.

Reply:
> and delay assigning it until later down when we actual calculate the JS filename (see js_target = .. in phase_final_emitting.. I would just set it then to the same value is js_target).

Hm, I can't actually delay it, because by that time all the macro has been already expanded to their values. Instead, I moved those `js_target ...` assignments to this phase, and reused computed value in the `phase_final_emitting`. I hope this is fine - it seems to work.

Btw, `EXPORT_ES6` is currently not compatible with `.html` outputs anyway (due to #7950 + #10086), and for `.wasm` outputs it's a no-op, so this might be not that big of a deal today, but worth fixing anyway.